runtime/src/main/java/org/corfudb/util/serializer/DynamicProtobufSerializer.java

```diff
@@ -90,31 +92,28 @@ public DynamicProtobufSerializer(CorfuRuntime corfuRuntime) {
         ISerializer protobufSerializer = createProtobufSerializer();
         Serializers.registerSerializer(protobufSerializer);
 
-        // Open the Registry Table.
-        CorfuTable<TableName, CorfuRecord<TableDescriptors, TableMetadata>> corfuTable = corfuRuntime.getObjectsView()
+        // Open the Protobuf Descriptor Table.
+        CorfuTable<ProtobufFileName, CorfuRecord<ProtobufFileDescriptor, TableMetadata>> descriptorTable = corfuRuntime.getObjectsView()
```

Comment:
This will load the complete `fdProtoMap` map, but only lazily load `fileDescriptorMap`. 

Also, I don't think the serializer itself should do the db access. A higher level of the abstraction should query the DB and initialize the Serializer with the schemas.

Reply:
DynamicProtobufSerializer is only used for tooling and checkpointing. If we aggressively build the complete dependency tree even before opening the first table it would slow down tools like browser which only really want to open one table.
That is the reason for the lazy initialization of fileDescriptorMap.
The serializer's DB access is limited to the constructor. Moving it outside will only complicate the caller's logic who does not need to know how DynamicProtobufSerialization works.